src/frontend/sharedComponents/ProjectInviteBottomSheet/index.tsx

```diff
@@ -1,145 +1,230 @@
+import {BottomSheetModalProvider} from '@gorhom/bottom-sheet';
 import * as React from 'react';
+import {View} from 'react-native';
+import {InviteInternal, InviteRemovalReason} from '@mapeo/core/dist/invite-api';
+import {MapBuffers} from '@mapeo/core/dist/types';
+
+import {useApi} from '../../contexts/ApiContext';
+import {usePendingInvites} from '../../hooks/server/invites';
 import {BottomSheetModal, useBottomSheetModal} from '../BottomSheetModal';
-import {
-  useAcceptInvite,
-  usePendingInvites,
-  useRejectInvite,
-} from '../../hooks/server/invites';
-import {useProjectInvitesListener} from '../../hooks/useProjectInvitesListener';
-import {NewInviteBottomSheetContent} from './NewInviteBottomSheetContent';
-import {InviteSuccessBottomSheetContent} from './InviteSuccessBottomSheetContent';
-import {InviteCanceledBottomSheetContent} from './InviteCanceledBottomSheetContent';
-import {useAllProjects} from '../../hooks/server/projects';
 import {LeaveProjectModalContent} from '../LeaveProjectModalContent';
+import {InviteBottomSheetContent} from './InviteBottomSheetContent';
 
-export type LeaveProjectModalState = 'AlreadyOnProj' | 'LeaveProj';
+export type SessionInvite =
+  | {
+      status: 'pending';
+      invite: MapBuffers<InviteInternal>;
+    }
+  | {
+      status: 'removed';
+      invite: MapBuffers<InviteInternal>;
+      removalReason: InviteRemovalReason;
+    };
 
 export const ProjectInviteBottomSheet = ({
   enabledForCurrentScreen,
 }: {
   enabledForCurrentScreen: boolean;
 }) => {
-  const {
-    sheetRef: inviteRef,
-    isOpen: inviteIsOpen,
-    closeSheet: closeInviteSheet,
-    openSheet: openInviteSheet,
-  } = useBottomSheetModal({
+  const inviteBottomSheet = useBottomSheetModal({
     openOnMount: false,
   });
 
-  const {
-    sheetRef: leaveRef,
-    isOpen: leaveIsOpen,
-    closeSheet: closeLeaveSheet,
-    openSheet: openLeaveSheet,
-  } = useBottomSheetModal({
+  const leaveProjectSheet = useBottomSheetModal({
     openOnMount: false,
   });
-  const invites = usePendingInvites().data.sort(
-    (a, b) => a.receivedAt - b.receivedAt,
-  );
 
-  const projects = useAllProjects();
+  // Ideally not needed but this is used for handling a state coordination complexity after rejecting an invite.
+  // There's a moment in time where the pending invites query properly removes the rejected invite
+  // but the data from useRemovedInvites() does not yet have the invite that was rejected.
+  // Without this, the `displayedInviteId` will again be set to this rejected invite which is not desirable, as it will cause the invite bottom sheet
+  // to open again unintentionally and run into an (ideally) impossible state, resulting in subsequent invites not opening the sheet as expected.
+  const lastRejectedInviteIdRef = React.useRef<string | undefined>();
```

Comment:
not proud of this one, but it was the least invasive way of solving a rather tricky state-related issue. Hard to describe in words but hoping the comment helps a little. Basically dealing with asynchronous state updates as a result of cache invalidation after rejecting an invite does not play well with some of the `useEffects()` in this component (which are unfortunately kind of necessary)